### DISCUSSION TITLE
I can't get through the instructions

### DISCUSSION THREAD

**Discussion by rocco-zhang** [FIRST-Tech-Challenge/fmltc#295] [Q&A]
I just spent like three hours figuring out why things weren't working and I eventually figured out it was because the entire readme was written for UNIX systems. I'd like to make a suggestion to inform readers of this and possibly direct them to or directly show them how to install a UNIX emulator like WSL for Windows.

**Comment by rocco-zhang** [comment]
After looking around a bit I saw someone else used the git bash terminal instead of WSL which I found works as well. Either way though, I'm stuck because I can't get past the step about setting the project ID and origin. After creating the app, the website shows me this: 
![image](https://user-images.githubusercontent.com/87133549/212580984-c2fdcbba-980c-4a86-8026-a4c8250aab1b.png)
However when I try to run gcloud app deploy I get an error which is this:
```
$ gcloud app deploy
Services to deploy:

descriptor:                  [D:\ftc\fmltc\build.gradle]
source:                      [D:\ftc\fmltc]
target project:              [fmltc00]
target service:              [default]
target version:              [20230115t204110]
target url:                  [https://fmltc00.uk.r.appspot.com]
target service account:      [App Engine default service account]


Do you want to continue (Y/n)?

Beginning deployment of service [default]...
#============================================================#
#= Uploading 0 files to Google Cloud Storage                =#
#============================================================#
File upload done.
Updating service [default]...
...........................................................................................................................failed.
ERROR: (gcloud.app.deploy) Error Response: [9] Cloud build 372cee10-8b7e-474c-a928-9a7bf601d024 status: FAILURE
/usr/bin/env: 'sh\r': No such file or directory
Full build logs: https://console.cloud.google.com/cloud-build/builds;region=us-east4/372cee10-8b7e-474c-a928-9a7bf601d024?project=339908667918
```
I've tried deleting and recloning this repo however I still run into this problem.

**Comment by rocco-zhang** [comment]
Even if I manually enter in an origin and deploy everything, I still run into issues. I'm unable to deploy the docker, here is the output:
```
root@LAPTOP-7C1286EL:/mnt/d/ftc/fmltc# scripts/deploy_docker_image.sh
2021_11_25: Pulling from lizlooney/object_detection
Digest: sha256:89fba7b53819d858d864628d9fc27df112fefffee54a47b9344e64dfc48953e0
Status: Image is up to date for ghcr.io/lizlooney/object_detection:2021_11_25
ghcr.io/lizlooney/object_detection:2021_11_25
The push refers to repository [gcr.io/fmltc00/object_detection]
b61f41524843: Preparing
e85bb5ee3500: Preparing
f6eae01ec4cc: Preparing
6625a1bb5cc7: Preparing
b1c5544f002d: Preparing
436ed910618c: Waiting
7c1d1709929f: Waiting
b9afb0403566: Waiting
c82fb73a41a4: Waiting
827a97679dff: Waiting
20c97f888e8c: Waiting
32cec01d1c97: Waiting
c47958f555a0: Waiting
c74eee0ce499: Waiting
31ff8bfc604d: Waiting
7979309529d3: Waiting
2f22a2e267f8: Waiting
8662098b1dc9: Waiting
c69ef24a9cfd: Waiting
9557d05771c8: Waiting
a5b452fa3051: Waiting
b5616a98a776: Waiting
4942a1abcbfa: Waiting
unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication
```
Without having the docker deployed, I tried to open the app and sign in and I get this: 
![image](https://user-images.githubusercontent.com/87133549/212587314-21d71f7d-a6b0-4ebf-84e3-e37428d76c13.png)
Which I'm not sure is a consequence of not deploying the docker or something else.

**Comment by cmacfarl** [comment]
The red text in your image indicates that it can't find the teams file.

Did you do this step?

https://github.com/FIRST-Tech-Challenge/fmltc#create-and-upload-the-team_infoteams-file

If so, can you confirm on the cloud console that it's actually where the software expects it to be?

**Comment by rocco-zhang** [comment]
Upon following the directions earlier and going to link that the step shows, my blob bucket looks like this:  (I already did this step)
![image](https://user-images.githubusercontent.com/87133549/212764653-0673d4fe-9be1-404e-8924-ff4adf93ea97.png)
You should be able to see that under [project-id]-blobs there is team_info and under that teams.txt.

**Comment by rocco-zhang** [comment]
I even tried making some other folders like o/team_info/teams.txt and I also noticed the error message says no such object .../teams so maybe its looking for a folder instead of a file but I'm not too sure and nothing has worked yet. However, my app engine works now for some reason and the page looks like this: 
![image](https://user-images.githubusercontent.com/87133549/212765691-fb91793f-f592-432a-a5ad-b9de949f7d84.png)
 I still run into the same error and I'm also not too sure why the URL says .uk.r. when the region is us-east4.
